internal/extramanifest/policy.go

```diff
@@ -0,0 +1,125 @@
+package extramanifest
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"log"
+	"sort"
+	"strconv"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	policyv1 "open-cluster-management.io/config-policy-controller/api/v1"
+	policiesv1 "open-cluster-management.io/governance-policy-propagator/api/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//+kubebuilder:rbac:groups=policy.open-cluster-management.io,resources=policies,verbs=get;list
+
+const ztpDeployWaveAnnotation = "ran.openshift.io/ztp-deploy-wave"
+
+// GetPolicies gets the policies matching the labels from the namespace
+func (h *EMHandler) GetPolicies(ctx context.Context, labels map[string]string) ([]*policiesv1.Policy, error) {
+	listOpts := []client.ListOption{
+		client.MatchingLabels(labels),
+	}
+
+	policies := &policiesv1.PolicyList{}
+	if err := h.Client.List(ctx, policies, listOpts...); err != nil {
+		return nil, err
+	}
+
+	var policyWaveMap = make(map[*policiesv1.Policy]int)
+
+	for _, policy := range policies.Items {
+		// Ignore policies with remediationAction enforce
+		if strings.EqualFold(string(policy.Spec.RemediationAction), "enforce") {
+			h.Log.Info("Ignoring policy " + policy.Name + " with remediationAction enforce")
+			continue
+		}
+
+		deployWave, found := policy.GetAnnotations()[ztpDeployWaveAnnotation]
+		if found {
+			deployWaveInt, err := strconv.Atoi(deployWave)
+			if err != nil {
+				// err convert from string to int
+				h.Log.Error(err, "%s in policy %s is not an interger: %s", ztpDeployWaveAnnotation, policy.GetName())
```

Comment:
```suggestion
				h.Log.Error(err, "%s in policy %s is not an integer: %s", ztpDeployWaveAnnotation, policy.GetName())
```

Reply:
Also this is `Error`, not some form of `Errorf` - as in, it accepts a list of "keys" and "values" (keys in even positions, values in odd positions), not merely format string parameters

Reply:
There is a plan to replace logr with logrus. But I will fix it anyways.

Reply:
I don't think we can replace logr completely. controller-sdk uses logr.

Reply:
Migration to logrus tracked under  CNF-10370